src/Core/Types/Types/ObjectType.cs

```diff
@@ -94,6 +97,44 @@ public bool IsOfType(IResolverContext context, object resolverResult)
                 CompleteIsOfType(context);
                 FieldInitHelper.CompleteFields(context, definition, Fields);
             }
+
+            InitializeMetaData(context);
```

Comment:
lets move this into the if clause and get rid of Fields != null since if validate fields is false a schema exception will prevent schema creation.